test/hook_permissions.py

```diff
@@ -0,0 +1,140 @@
+            hook_detected = False
+        if push_access and hook_detected or admin_access:
+            return True
+        if push_access and not hook_detected:
+            print('Warning: Push access detected but unable to verify manual hook '
+                'configuration for repo [ %s ]. Please visit ' % repo.full_name +
+                'http://wiki.ros.org/buildfarm/Pull%20request%20testing '
+                'and make sure hooks are setup.')
+            if strict:
+                return False
+            else:
+                errors.append('Warning: Push access detected but unable to verify manual hook '
+                    'configuration for repo [ %s ]. Please visit ' % repo.full_name +
+                    'http://wiki.ros.org/buildfarm/Pull%20request%20testing '
+                    'and make sure hooks are setup.')
+                return True
+
+def check_hooks_on_repo(user, repo, errors, hook_user='ros-pull-request-builder', callback_url='http://build.ros.org/ghprbhook/', token=None, strict=False):
+
+    ghprb_detector = GHPRBHookDetector(hook_user, token, callback_url)
+
+    test_repo = ghprb_detector.get_repo(user, repo)
+
+    if test_repo:
+        hooks_ok = ghprb_detector.check_repo_for_access(test_repo, errors, strict=strict)
+        if hooks_ok:
+            print('Passed ghprb_detector check for hooks access'
+                  ' for repo [ %s ]' % test_repo.full_name)
+            return True
+        else:
+            print('ERROR: Not enough permissions to setup pull request'
+                  ' builds for repo [ %s ] ' % (test_repo.full_name) +
+                  'Please see http://wiki.ros.org/buildfarm/Pull%20request%20testing'
+                  )
+            return False
+    else:
+        print(
+            'ERROR: No github repository found at %s/%s' % (args.user, args.repo)
+        )
+        return False
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Test for ros-pull-request-builder access')
+    parser.add_argument('user', type=str)
+    parser.add_argument('repo', type=str)
+    parser.add_argument('--callback-url', type=str,
+        default='http://build.ros.org/ghprbhook/')
+    parser.add_argument('--hook-user', type=str,
+        default='ros-pull-request-builder')
+    parser.add_argument('--password-env', type=str,
+        default='ROSGHPRB_TOKEN')
+
+    args = parser.parse_args()
+
+    password = os.getenv(args.password_env)
+    if not password:
+        parser.error(
+            'OAUTH Token with hook and organization read access'
+            'required in ROSGHPRB_TOKEN environment variable')
+    errors = []
+    result = check_hooks_on_repo(
+        args.user,
+        args.repo,
+        errors,
+        args.hook_user,
+        args.callback_url,
+        password)
+    for e in errors:
+        print(e)
+    return result
+
+
+if __name__ == '__main__':
+    sys.exit(main() == False)
```

Comment:
A boolean return from a `main` function is rather uncommon. I would suggest a value which can directly be passed to `sys.exit`.